cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceContextConfig.java

```diff
@@ -177,13 +179,14 @@ WorkflowProgramInfo getWorkflowProgramInfo() {
 
   private void setPlugins(Map<String, Plugin> plugins) {
     hConf.set(HCONF_ATTR_PLUGINS, GSON.toJson(plugins, PLUGIN_MAP_TYPE));
+    cConf.set(HCONF_ATTR_PLUGINS, GSON.toJson(plugins, PLUGIN_MAP_TYPE));
```

Comment:
Why need to store in both cConf and hConf?

Reply:
The right fix should be not doing variable substitution when getting the plugin attributes from the configuration.